File: src/main/java/net/zoe/corporation/structures/commands/CommandOptions.java
package net.zoe.corporation.structures.commands;

import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
import net.zoe.corporation.Config;
import net.zoe.corporation.utilities.ZoeTranslateGetter;

public class CommandOptions {
    public CommandOptions(Command cmd, GuildMessageReceivedEvent event, ZoeTranslateGetter t) {
        String author = event.getAuthor().getId();
        // Mr Gambiarras 
        if(cmd.devsOnly && 
        author != Config.owners[0] 
        && author != Config.owners[1] 
        && author != Config.owners[2]) {
            event.getChannel().sendMessage("macaco").queue();
        }
    }
}

File: src/main/kotlin/com/github/mrdroox/projects/utilities/commands/CommandOptions.kt
package com.github.mrdroox.projects.utilities.commands

import com.github.mrdroox.projects.utilities.Config
import com.github.mrdroox.projects.utilities.commands.enumerator.CommandCategory
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent

class CommandOptions(private val ctx: CommandContext, private val command: Command, private val AuthorID: String, private val event: GuildMessageReceivedEvent) {
    init { context() }

    private fun context() {
        if(command.category === CommandCategory.DEVELOPER && AuthorID !== Config.get("owner")) return
        if(command.category === CommandCategory.RIORU_ROOM && event.guild.id !== Config.get("support_server_id")) {
            ctx.sendMessage("CommandOptions:utils:")
            return
        }

        if(command.argsSize !== null && ctx.getArgs().size < command.argsSize!!) {
            ctx.sendMessage("")
            return
        }
    }
}